centreon/src/Core/Dashboard/Playlist/Application/UseCase/SharePlaylist/SharePlaylist.php

```diff
@@ -0,0 +1,225 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Dashboard\Playlist\Application\UseCase\SharePlaylist;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\Repository\Interfaces\DataStorageEngineInterface;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Application\Common\UseCase\NoContentResponse;
+use Core\Application\Common\UseCase\NotFoundResponse;
+use Core\Contact\Application\Repository\ReadContactGroupRepositoryInterface;
+use Core\Contact\Application\Repository\ReadContactRepositoryInterface;
+use Core\Contact\Domain\Model\ContactGroup;
+use Core\Dashboard\Domain\Model\DashboardRights;
+use Core\Dashboard\Playlist\Application\Exception\PlaylistException;
+use Core\Dashboard\Playlist\Application\Repository\WritePlaylistShareRepositoryInterface;
+
+final class SharePlaylist
+{
+    use LoggerTrait;
+
+    /**
+     * @param DashboardRights $rights
+     * @param SharePlaylistValidator $validator
+     * @param ContactInterface $user
+     * @param WritePlaylistShareRepositoryInterface $shareRepository
+     * @param DataStorageEngineInterface $dataStorageEngine
+     * @param ReadContactGroupRepositoryInterface $contactGroupRepository
+     * @param ReadContactRepositoryInterface $contactRepository
+     */
+    public function __construct(
+        private readonly DashboardRights $rights,
+        private readonly SharePlaylistValidator $validator,
+        private readonly ContactInterface $user,
+        private readonly WritePlaylistShareRepositoryInterface $shareRepository,
+        private readonly DataStorageEngineInterface $dataStorageEngine,
+        private readonly ReadContactGroupRepositoryInterface $contactGroupRepository,
+        private readonly ReadContactRepositoryInterface $contactRepository,
+    ) {
+    }
+
+    /**
+     * @param int $playlistId
+     * @param SharePlaylistRequest $request
+     * @param SharePlaylistPresenterInterface $presenter
+     */
+    public function __invoke(
+        int $playlistId,
+        SharePlaylistRequest $request,
+        SharePlaylistPresenterInterface $presenter
+    ): void {
+        try {
+            if (! $this->rights->canCreate()) {
+                $this->error('Access Forbidden');
```

Comment:
IMHO this could be rephrased to give more context something like
`User does not have sufficient rigths to create a Playlist`